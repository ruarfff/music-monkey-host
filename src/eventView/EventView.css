.EventView-title {
  padding-left: 1em;
}
