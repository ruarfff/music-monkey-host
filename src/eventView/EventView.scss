.EventView-title {
  padding-left: 1em;
}

.eventViewWrapper {
  height: 100%;
  margin-left: -10px;
}
