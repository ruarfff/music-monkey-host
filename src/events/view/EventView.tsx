import DeleteIcon from '@material-ui/icons/Delete'
import EditIcon from '@material-ui/icons/ModeEdit'
import * as classNames from 'classnames'
import { Typography } from 'material-ui'
import AppBar from 'material-ui/AppBar'
import Button from 'material-ui/Button'
import red from 'material-ui/colors/red'
import Grid from 'material-ui/Grid'
import { Theme, WithStyles, withStyles } from 'material-ui/styles'
import Tabs, { Tab } from 'material-ui/Tabs'
import Zoom from 'material-ui/transitions/Zoom'
import * as React from 'react'
import * as CopyToClipboard from 'react-copy-to-clipboard'
import lifecycle from 'react-pure-lifecycle'
import { RouteComponentProps } from 'react-router'
import SwipeableViews from 'react-swipeable-views'
import Swal from 'sweetalert2'
import withReactContent from 'sweetalert2-react-content'
import IAction from '../../Action'
import LoadingSpinner from '../../loading/LoadingSpinner'
import LinkButton from '../../util/LinkButton'
import EventFetchError from '../EventFetchError'
import IEvent from '../IEvent'
import PreGameView from '../preGame/PreGameViewContainer'
import EventDetails from './EventDetails'

interface IEventViewProps extends RouteComponentProps<any> {
  error?: Error
  event: IEvent
  loading?: boolean
  eventTabIndex: number
  theme: any
  deleteSelected: boolean
  deleteSuccess: boolean
  deleteFailed: boolean
  getEventById(eventId: string): IAction
  onEventTabIndexChange(index: number): IAction
  onEventDeleteSelected(): IAction
  onEventDeleteClosed(): IAction
  deleteEvent(event: IEvent): IAction
  onDeleteAknowledged(): IAction
}
type PropsWithStyles = IEventViewProps &
  WithStyles<'root' | 'button' | 'rightIcon' | 'deleteButton'>

const style = (theme: Theme) => ({
  root: {
    padding: theme.spacing.unit,
    backgroundColor: theme.palette.background.default,
    color: theme.palette.primary.main
  },
  button: {
    margin: theme.spacing.unit
  },
  rightIcon: {
    marginLeft: theme.spacing.unit
  },
  deleteButton: {
    color: theme.palette.getContrastText(red[500]),
    backgroundColor: red[500],
    '&:hover': {
      backgroundColor: red[700]
    }
  }
})

const SweetAlert = withReactContent(Swal) as any

const callGetEventById = (props: PropsWithStyles) => {
  return () => props.getEventById(props.match.params.eventId)
}

const componentDidMount = (props: PropsWithStyles) => {
  callGetEventById(props)()
}

function TabContainer({ children, dir }: any) {
  return (
    <Typography component="div" dir={dir} style={{ padding: 8 * 3 }}>
      {children}
    </Typography>
  )
}

const handleTabChange = (props: IEventViewProps) => (
  event: any,
  index: number
) => {
  props.onEventTabIndexChange(index)
}

const showDeleteSuccess = (props: PropsWithStyles) => {
  SweetAlert.fire({
    title: 'Event Deleted',
    type: 'success'
  }).then(() => {
    props.onDeleteAknowledged()
    props.onEventDeleteClosed()
  })
}

const showDeleteFailed = (props: PropsWithStyles) => {
  SweetAlert.fire({
    title: "Clouldn't delete Event",
    text: 'Sorry. An error occurred when trying to delete this Event.',
    type: 'error'
  }).then(() => {
    props.onEventDeleteClosed()
  })
}

const handleDeleteSelected = (props: IEventViewProps) => () => {
  SweetAlert.fire({
    title: 'Are you sure?',
    text: 'This will completely remove this event',
    type: 'warning',
    showCancelButton: true,
    confirmButtonColor: '#3085d6',
    cancelButtonColor: '#d33',
    confirmButtonText: 'Yes, delete it!'
  }).then((result: any) => {
    if (result.value) {
      props.deleteEvent(props.event)
    }
    props.onEventDeleteClosed()
  })
}

const handleCopyToClipboard = (props: PropsWithStyles) => {
  return () => true
}

const renderEventView = (props: PropsWithStyles) => (
  <Grid container={true} spacing={16}>
    <Grid item={true} xs={12} sm={6}>
      <Typography variant="display3" noWrap={true}>
        {props.event && props.event.name}
      </Typography>
    </Grid>
    <Grid item={true} xs={12} sm={2}>
      <Typography variant="display4" noWrap={true}>
        {props.event &&
          props.event.inviteLink && (
            <CopyToClipboard
              text={props.event.inviteLink}
              onCopy={handleCopyToClipboard(props)}
            >
              <button>Copy to clipboard with button</button>
            </CopyToClipboard>
          )}
      </Typography>
    </Grid>
    <Grid item={true} xs={12} sm={4}>
      <LinkButton
        className={props.classes.button}
        variant="raised"
        color="secondary"
        to={props.location.pathname + '/edit'}
      >
        Edit
        <EditIcon className={props.classes.rightIcon} />
      </LinkButton>
      <Button
        className={classNames(props.classes.button, props.classes.deleteButton)}
        variant="raised"
        color="secondary"
        onClick={handleDeleteSelected(props)}
      >
        Delete
        <DeleteIcon className={props.classes.rightIcon} />
      </Button>
    </Grid>

    <Grid item={true} xs={12}>
      <AppBar position="static" color="default">
        <Tabs
          value={props.eventTabIndex}
          onChange={handleTabChange(props)}
          indicatorColor="primary"
          textColor="primary"
          scrollable={true}
          scrollButtons="auto"
        >
          <Tab label="Pre-Game" />
          <Tab label="In-Game" />
          <Tab label="Event" />
        </Tabs>
      </AppBar>
      <SwipeableViews
        axis={props.theme.direction === 'rtl' ? 'x-reverse' : 'x'}
        index={props.eventTabIndex}
        onChangeIndex={props.onEventTabIndexChange}
      >
        <TabContainer dir={props.theme.direction}>
          <PreGameView />
        </TabContainer>
        <TabContainer dir={props.theme.direction}>
          Nothing Here Yet
        </TabContainer>
        <TabContainer dir={props.theme.direction}>
          <EventDetails event={props.event} />
        </TabContainer>
      </SwipeableViews>
    </Grid>
  </Grid>
)

const EventView: React.SFC<PropsWithStyles> = (props: PropsWithStyles) => (
  <div>
    {props.loading && <LoadingSpinner />}
    {!props.loading &&
      props.error && <EventFetchError onTryAgain={callGetEventById(props)} />}
    {!!props.event && (
      <Zoom in={!props.loading && !!props.event}>{renderEventView(props)}</Zoom>
    )}
    {props.deleteFailed && showDeleteFailed(props)}
    {props.deleteSuccess && showDeleteSuccess(props)}
  </div>
)

export default lifecycle({
  componentDidMount
})(withStyles(style, { withTheme: true })<IEventViewProps>(EventView))
