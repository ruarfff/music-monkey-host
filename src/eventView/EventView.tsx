import AppBar from '@material-ui/core/AppBar/AppBar'
import { WithStyles } from '@material-ui/core/es'
import Grid from '@material-ui/core/Grid/Grid'
import withStyle from '@material-ui/core/styles/withStyles'
import Tab from '@material-ui/core/Tab/Tab'
import Tabs from '@material-ui/core/Tabs/Tabs'
import Typography from '@material-ui/core/Typography/Typography'
import { isEmpty } from 'lodash'
import * as React from 'react'
import { RouteComponentProps } from 'react-router'
import InviteCopyAlert from '../components/InviteLink/InviteCopyAlert'
import EventFetchError from '../event/EventFetchError'
import IEvent from '../event/IEvent'
import IAction from '../IAction'
import LoadingSpinner from '../loading/LoadingSpinner'
import {
  onGuestUpdate,
  subscribeToSuggestionsAccepted,
  subscribeToVotesModified
} from '../notification'
import EventGuests from './components/Guests/EventGuestsContainer'
import EventPlaylistView from './components/Playlist/EventPlaylistViewContainer'
import EventSummaryView from './components/Summary/EventSummaryViewContainer'
import './EventView.scss'

const decorated = withStyle(() => ({
  tabContainer: {
    background: 'transparent',
    boxShadow: 'none',
    borderBottom: '1px solid #d6d6d6',
    marginBottom: '10px'
  },
  tab: {
    color: '#979797'
  },
  tabs: {
    color: '#AF00FF!important',
    borderBottom: '1px solid #979797'
  },
  TabIndicator: {
    backgroundColor: '#AF00FF!important',
    height: '1px'
  },
  content: {
    paddingTop: 0,
    height: '100%'
  }
}))

interface IEventViewState {
  tabIndex: number
}

interface IEventViewProps extends RouteComponentProps<any> {
  error: Error
  event: IEvent
  loading: boolean
  copiedToClipboard: boolean
  getEventById(eventId: string): IAction
  copyEventInvite(): IAction
  acknowledgeEventInviteCopied(): IAction
  getEventSuggestions(eventId: string): IAction
  fetchEventVotes(eventId: string): IAction
  getEventByIdNoLoading(eventId: string): IAction
}

function TabContainer({ children, dir }: any) {
  return (
    <Typography style={{ height: '100%' }} component="div" dir={dir}>
      {children}
    </Typography>
  )
}

class EventView extends React.Component<
  IEventViewProps & WithStyles,
  IEventViewState
> {
  public state = {
    tabIndex: 0
  }

  public componentDidMount() {
    const eventId = this.props.match.params.eventId

    this.props.getEventById(eventId)
    this.props.fetchEventVotes(eventId)
    this.props.getEventSuggestions(eventId)
  }

  public componentWillReceiveProps(newProps: IEventViewProps) {
    const eventId = newProps.match.params.eventId
    if (isEmpty(newProps.event) && !newProps.loading) {
      this.props.getEventById(eventId)
      this.props.fetchEventVotes(eventId)
      this.props.getEventSuggestions(eventId)
    }
  }

  public componentWillUpdate() {
    const eventId = this.props.match.params.eventId
    subscribeToSuggestionsAccepted(eventId, this.handleSuggestionNotification)
    subscribeToVotesModified(eventId, this.handleEventVotesModified)
    onGuestUpdate(eventId, this.handleUpdateGuests)
  }

  public render() {
    const {
      loading,
      error,
      event,
      copiedToClipboard,
      acknowledgeEventInviteCopied
    } = this.props

    const shouldShowEvent: boolean = !loading && !isEmpty(event)

    return (
      <div className="eventViewWrapper">
        {loading && <LoadingSpinner />}
        {loading && !isEmpty(error) && (
          <EventFetchError onTryAgain={this.handleGetEvent} />
        )}
        {shouldShowEvent && this.renderEventView()}
        {copiedToClipboard && (
          <InviteCopyAlert
            message="Copied to Clipboard"
            onClose={acknowledgeEventInviteCopied}
          />
        )}
      </div>
    )
  }

  private renderEventView = () => {
    const { tabIndex } = this.state
    const { classes } = this.props

    return (
      <Grid className={classes.content} container={true} spacing={16}>
        <Grid className={classes.content} item={true} xs={12}>
          <AppBar
            position="static"
            color="primary"
            className={classes.tabContainer}
          >
            <Tabs
              value={tabIndex}
              onChange={this.handleTabChange}
              indicatorColor={'primary'}
              textColor={'primary'}
              TabIndicatorProps={{ className: classes.TabIndicator }}
              centered={true}
              className={classes.tabs}
              fullWidth={true}
            >
              <Tab className={classes.tab} label="Event Summary" />
              <Tab className={classes.tab} label="Playlist" />
              <Tab className={classes.tab} label="Guest List" />
            </Tabs>
          </AppBar>
          {tabIndex === 0 && (
            <TabContainer className={classes.content} dir={'x'}>
              <EventSummaryView />
            </TabContainer>
          )}
          {tabIndex === 1 && (
            <TabContainer className={classes.content} dir={'x'}>
              <EventPlaylistView />
            </TabContainer>
          )}
          {tabIndex === 2 && (
            <TabContainer className={classes.content} dir={'x'}>
              <EventGuests />
            </TabContainer>
          )}
        </Grid>
      </Grid>
    )
  }

  private handleGetEvent() {
    this.props.getEventById(this.props.match.params.eventId)
  }

  private handleTabChange = (event: any, index: number) => {
    this.setState({ tabIndex: index })
  }

  private handleUpdateGuests = () => {
    const eventId = this.props.match.params.eventId
    if (eventId) {
      this.props.getEventById(eventId)
    }
  }

  private handleSuggestionNotification = () => {
    const eventId = this.props.match.params.eventId
    if (eventId) {
      this.props.getEventSuggestions(eventId)
    }
  }

  private handleEventVotesModified = () => {
    const eventId = this.props.match.params.eventId
    const { event } = this.props
    if (eventId) {
      this.props.fetchEventVotes(eventId)
    }
    if (!isEmpty(event) && event.settings.dynamicVotingEnabled) {
      this.props.getEventByIdNoLoading(eventId)
    }
  }
}

export default decorated(EventView)
