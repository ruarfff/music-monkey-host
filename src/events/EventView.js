import React, { Component } from 'react'
import './EventView.css'

class EventView extends Component {
  render() {
    return (
      <div>
        <h1>Event Veiw!!!</h1>
      </div>
    )
  }
}

EventView.propTypes = {}

export default EventView
