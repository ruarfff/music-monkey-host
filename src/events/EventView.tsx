import { withStyles, WithStyles } from 'material-ui/styles'
import * as React from 'react'
import Action from '../Action'
import LoadingSpinner from '../loading/LoadingSpinner'
import { IEvent } from './EventModel'

import './EventView.css'

const decorate = withStyles(({ palette, spacing }) => ({
  root: {
    padding: spacing.unit,
    backgroundColor: palette.background.default,
    color: palette.primary.main
  }
}))

interface IEventViewProps {
  error: Error
  event: IEvent
  loading: boolean
  match: any
  getEventById(eventId: string): Action
}

const renderError = (error: Error) => (
  <h1>ERORORORORO {JSON.stringify(error)}</h1>
)

const renderEvent = (event: IEvent) => <div>{event.eventId}</div>

const EventView = decorate(
  class extends React.Component<IEventViewProps & WithStyles<'root'>, {}> {
    public componentDidMount() {
      this.props.getEventById(this.props.match.params.eventId)
    }

    public render() {
      const { loading, error, event } = this.props
      return (
        <div>
          {loading && <LoadingSpinner />}
          {(!loading && error) && renderError(error)}
          {(!loading && event) && renderEvent(event)}
        </div>
      )
    }
  }
)

export default EventView
