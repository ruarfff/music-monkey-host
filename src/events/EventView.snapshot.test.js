import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import EventView from './EventView'

describe('EventView Snapshot', () => {
  it('has not changed unexpectedly', () => {
    const component = renderer.create(
      <MemoryRouter>
        <EventView />
      </MemoryRouter>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
