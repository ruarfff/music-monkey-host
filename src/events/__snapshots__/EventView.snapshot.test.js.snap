// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventView Snapshot has not changed unexpectedly 1`] = `
<div>
  <h1>
    Event Veiw!!!
  </h1>
</div>
`;
